FinancialTracker/ViewControllers/UsersTableViewController.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  UsersViewController.swift
+//  FinancialTracker
+//
+//  Created by Boyan Pavlov on 22.02.22.
+//
+
+import UIKit
+
+class UsersTableViewController: UITableViewController {
+    var usersEmails: [String]?
+    var authManager: AuthManager?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        authManager?.getAllUsers(completionHandler: { firebaseError, emails in
+            if let firebaseError = firebaseError {
+                assertionFailure(firebaseError.localizedDescription)
+                return
+            }
+            self.usersEmails = emails
+            
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        })
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return usersEmails?.count ?? 0
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "UserTableViewCell", for: indexPath)
+        
+        cell.textLabel?.text = usersEmails?[indexPath.row]
+        cell.accessoryType = .disclosureIndicator
+        
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let email = usersEmails?[indexPath.row] else {
+            assertionFailure("Non exisiting email.")
+            return
+        }
+        
+        let alertVC = UIAlertController(title: "Enter amount", message: email, preferredStyle: .alert)
+        
+        alertVC.addTextField { textField in
+            textField.placeholder = "Amount"
+        }
+
+        alertVC.addAction(UIAlertAction(title: "Request", style: .default, handler: { _ in
+            guard let amount = alertVC.textFields?[0].text else {
+                assertionFailure("Empty textField.")
+                return
+            }
+
+            guard let amountValue = Double(amount) else {
+                assertionFailure("Enter a number.")
+                return
+            }
+
+            self.requestOrSend(email: email, amount: amountValue, transferType: TransferType.request)
+        }))
+
+        alertVC.addAction(UIAlertAction(title: "Send", style: .default, handler: { _ in
```

Comment:
put the UIAlertAction in its own property - it will help the code look more organised.